### PR TITLE
Cleanup of Performance Improvements for ODE Conditioning

### PR DESCRIPTION
**Pull request by azane**
Blocking #219 

Does not implement handler composition tests. That's captured in issue #149.

### CODE REVIEW

File: tests/dynamical/dynamical_fixtures.py [hunk 1]
```diff
@@ -24,8 +24,6 @@ def observation(self, X: State[torch.Tensor]):
         S_obs = pyro.sample("S_obs", Normal(X.S, 1))
         I_obs = pyro.sample("I_obs", Normal(X.I, 1))
         R_obs = pyro.sample("R_obs", Normal(X.R, 1))
-        usa_expected_cost = torch.relu(S_obs + 2 * I_obs - R_obs)
```
Comment (azane) [inline tests/dynamical/dynamical_fixtures.py:27]: This just wasn't being used by any tests.

File: tests/dynamical/test_static_observation.py [hunk 2]
```diff
@@ -30,29 +29,41 @@
...
+        with PointObservation(time=3.1, data=data2):
+            with PointObservation(time=2.9, data=data1):
                 result = simulate(model, init_state, tspan)
 
     assert result.S.shape[0] == 5
     assert result.I.shape[0] == 5
     assert result.R.shape[0] == 5
 
 
+def _get_compatible_observations(obs_handler, time, data):
```
Comment (azane) [inline tests/dynamical/test_static_observation.py:49]: This just lets us use the same code to test both `PointObservation` and the vectorized non-interrupting version.

File: tests/dynamical/test_static_observation.py [hunk 3]
```diff
@@ -123,62 +128,40 @@ def conditioned_sir():
         svi.step()
 
 
-@pytest.mark.parametrize("model", [SimpleSIRDynamics()])
-def test_interrupting_and_non_interrupting_observation_equivalence(model):
```
Comment (azane) [inline tests/dynamical/test_static_observation.py:127]: Test is redundant w/ below.

File: tests/dynamical/test_static_observation.py [hunk 4]
```diff
@@ -123,62 +128,40 @@ def conditioned_sir():
...
-
-        with pyro.poutine.trace() as tr2:
-            with SimulatorEventLoop():
-                with NonInterruptingPointObservationArray(times=times, data=data):
-                    non_interrupting_ret = simulate(model, init_state, tspan)
-
-        logprobs.append([tr1.trace.log_prob_sum(), tr2.trace.log_prob_sum()])
+                        interrupting_ret = simulate(model, init_state, tspan)
 
-    # Randomness here is due to solver.
```
Comment (azane) [inline tests/dynamical/test_static_observation.py:174]: It wasn't due to solver, it was cz I wasn't conditioning on everything.

**Review by agrawalraj — APPROVED** [review]

**Pull request merged**